src/list.rs

```diff
@@ -53,24 +53,27 @@ impl<T: Value, N: Unsigned, U: UpdateMap<T>> List<T, N, U> {
         Self::try_from_iter(vec)
     }
 
-    pub(crate) fn from_parts(tree: Arc<Tree<T>>, depth: usize, length: Length) -> Self {
+    pub(crate) fn from_parts(tree: Arc<Tree<T>>, length: Length) -> Result<Self, Error> {
+        if length.as_usize() > N::to_usize() {
+            return Err(Error::BuilderFull);
```

Comment:
I think a different error would be clearer here, like `Error::InvalidLengthFromParts { len: usize, expected: usize }`.